element_array_ephys/export/nwb/nwb.py

```diff
@@ -0,0 +1,629 @@
+    """
+
+    # electrode id mapping
+    mapping = get_electrodes_mapping(nwbfile.electrodes)
+
+    units_table = pynwb.misc.Units(name=name, description=desc)
+    for additional_attribute in ["cluster_quality_label", "spike_depths"]:
+        units_table.add_column(
+            name=units_query.heading.attributes[additional_attribute].name,
+            description=units_query.heading.attributes[additional_attribute].comment,
+            index=additional_attribute == "spike_depths",
+        )
+
+    clustering_query = (
+        ephys.EphysRecording * ephys.ClusteringTask & session_key & paramset_record
+    )
+
+    for unit in tqdm(
+        (ephys.CuratedClustering.Unit & clustering_query).fetch(as_dict=True),
+        desc=f"creating units table for paramset {paramset_record['paramset_idx']}",
+    ):
+
+        probe_id, shank_num = (
+            probe.ProbeType.Electrode
+            * ephys.ProbeInsertion
+            * ephys.CuratedClustering.Unit
+            & {"unit": unit["unit"], "insertion_number": unit["insertion_number"]}
+        ).fetch1("probe", "shank")
+
+        waveform_mean = (
+            ephys.WaveformSet.PeakWaveform() & clustering_query & unit
+        ).fetch1("peak_electrode_waveform")
+
+        units_table.add_row(
+            id=unit["unit"],
+            electrodes=[mapping[(probe_id, unit["electrode"])]],
+            electrode_group=nwbfile.electrode_groups[
+                f"probe{probe_id}_shank{shank_num}"
+            ],
+            cluster_quality_label=unit["cluster_quality_label"],
+            spike_times=unit["spike_times"],
+            spike_depths=unit["spike_depths"],
+            waveform_mean=waveform_mean,
+        )
+
+    return units_table
+
+
+def add_ephys_units_to_nwb(
+    session_key: dict, nwbfile: pynwb.NWBFile, primary_clustering_paramset_idx: int = 0
+):
+    """
+    Add spiking data to NWBFile.
+
+    ephys.CuratedClustering.Unit::unit -> units.id
+    ephys.CuratedClustering.Unit::spike_times -> units["spike_times"]
+    ephys.CuratedClustering.Unit::spike_depths -> units["spike_depths"]
+    ephys.CuratedClustering.Unit::cluster_quality_label -> units["cluster_quality_label"]
+
+    ephys.WaveformSet.PeakWaveform::peak_electrode_waveform -> units["waveform_mean"]
+
+    Parameters
+    ----------
+    session_key: dict
+    nwbfile: pynwb.NWBFile
+    primary_clustering_paramset_idx: int, optional
+    """
+
+    if not ephys.ClusteringTask & session_key:
+        warnings.warn(f'No unit data exists for session:{session_key}')
+        return
+
+    if nwbfile.electrodes is None:
+        add_electrodes_to_nwb(session_key, nwbfile)
+
+    # add additional columns to the units table
+    units_query = ephys.CuratedClustering.Unit() & session_key
+
+    for paramset_record in (
+        ephys.ClusteringParamSet & ephys.CuratedClustering & session_key
+    ).fetch("paramset_idx", "clustering_method", "paramset_desc", as_dict=True):
```

Comment:
```suggestion
    for curation_record in (ephys.CuratedClustering & session_key).fetch('paramset_idx', 'curation_id', as_dict=True):
```

Reply:
Perhaps in this for loop we should iterate through the `ephys.CuratedClustering` table which will in effect iterate through all the parameter sets and manual curations for this `session_key`?

Reply:
@kabilar I don't really understand `ephys.CuratedClustering`. Are you saying that there are potentially multiple manual curations for a given paramset? When I run `ephys.CuratedClustering()`, I get

![image](https://user-images.githubusercontent.com/844306/161641030-4194c40a-c662-4a60-b69a-1abc0e5cb52f.png)

There are unique IDs for paramset: `paramset_idx`. But if there were multiple manual curations, how would those be differentiated without a unique identifier for those too? Since that was missing, I assume these tables where one-to-one, and it didn't really matter which one I iterated over.

Reply:
Ah yes, my mistake.  For this export function, we are using the `ephys_no_curation` module which doesn't allow for manual curations and thus does not have a `curation_id`.  Please ignore this suggestion.